crates/noirc_evaluator/src/ssa/opt/inlining.rs

```diff
@@ -35,8 +35,16 @@ impl Ssa {
     /// changes. This is because if the function's id later becomes known by a later
     /// pass, we would need to re-run all of inlining anyway to inline it, so we might
     /// as well save the work for later instead of performing it twice.
-    pub(crate) fn inline_functions(self) -> Ssa {
-        InlineContext::new(&self).inline_all(self)
+    pub(crate) fn inline_functions(mut self) -> Ssa {
+        self.functions = get_entry_point_functions(&self)
+            .into_iter()
+            .map(|entry_point| {
+                let new_function = InlineContext::new(&self, entry_point).inline_all(&self);
+                (entry_point, new_function)
+            })
+            .collect();
```

Comment:
This is the main change, we always set `self.functions` to only the inlined set of entry point functions. Previously, `self.functions` could contain every function before inlining + the inlined `main` function. Now by unconditionally setting this, it is expected that every function present before inlining is always removed or replaced.